### ISSUE TITLE
CLI deprecation false positive

### ISSUE THREAD

**Issue by emilk** [rust-lang/rust-clippy#6879]
I tried this:

`cargo clippy -- -D clippy::needless_pass_by_value`

and got this warning:

```
warning: lint name `needless_pass_by_value` is deprecated and does not have an effect anymore. Use: clippy::needless_pass_by_value
  |
  = note: requested on the command line with `-D needless_pass_by_value`
```

So the suggestion is to use exactly the format I am using.

`cargo clippy -- -D needless_pass_by_value` produces the same warning.

### Meta

- `cargo clippy -V`: clippy 0.0.212 (cb75ad5d 2021-02-10)
- `rustc -Vv`:
```
rustc 1.50.0 (cb75ad5db 2021-02-10)
binary: rustc
commit-hash: cb75ad5db02783e8b0222fee363c5f63f7e2cf5b
commit-date: 2021-02-10
host: x86_64-apple-darwin
release: 1.50.0
```

**Comment by giraffate** [comment]
I couldn't see that warning. Can you provide a minimal reproduction? FYI I tried codes below:
```rust
#![allow(dead_code)]
fn main() {}

fn foo(v: Vec<i32>) {
    assert_eq!(v.len(), 42);
}
```
and I get warnings below:
```
$ rustc +stable -V                           
rustc 1.50.0 (cb75ad5db 2021-02-10)
$ cargo +stable clippy -- -D clippy::needless_pass_by_value
    Checking issue6879 v0.1.0 (/Users/tnakata/workspace/rust-playground/issue6879)
error: this argument is passed by value, but not consumed in the function body
 --> src/main.rs:4:11
  |
4 | fn foo(v: Vec<i32>) {
  |           ^^^^^^^^ help: consider changing the type to: `&[i32]`
  |
  = note: requested on the command line with `-D clippy::needless-pass-by-value`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_pass_by_value

error: aborting due to previous error

error: could not compile `issue6879`

To learn more, run the command again with --verbose.
```

**Comment by emilk** [comment]
I fail to reproduce this myself now. WEIRD! I'll take better noted if it reappears.
